Gateway/Http/Client/TransactionDonate.php

```diff
@@ -63,17 +61,15 @@ public function __construct(
     public function placeRequest(TransferInterface $transferObject): array
     {
         $request = $transferObject->getBody();
-        $headers = $transferObject->getHeaders();
-
+        $requestOptions['headers'] = $this->requestHeaders($transferObject);
```

Comment:
Ideally `$transferObject->getHeaders()` should be sufficient to fetch the headers of the requests. Implementing an abstract class for an abstract method would increase the complexity of the design. Let's discuss this approach.